### DISCUSSION TITLE
What are the ways to explicitly control how the clock tree is created?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#1258] [General] [answered]
I was able to improve the clock tree of my design dramatically by increasing max routing layer from default M7 in ASAP7 to M9 for a design that has macros and where it routing the clock tree on top of the macros reduces latency and skew for the clock tree.

However, this is not an explicit instruction to the CTS algorithm, it is more freedom for CTS to create lower latency and skew.

```
# Need a layer to route clocks on
export MAX_ROUTING_LAYER       = M9
```

Q: Are there any explicit ways or other less explicit ways to control how the clock tree is generated?

**Comment by maliberty** [comment, marked as answer]
The CTS command has a lot of knobs but they are most useful with an autotuner.  I suspect you had more of a problem with routing detouring than cts itself.

**Reply by oharboe** [reply]
Indeed lots of options to `clock_tree_synthesis`, but none of them stand out as an obvious way to reduce latency and skew: https://github.com/The-OpenROAD-Project/OpenROAD/blob/65aff1f6e4163316511ae2a3266ca058f150c3e2/src/cts/src/TritonCTS.tcl#L89
